documentation/design-docs/diagnostics-client-library.md

```diff
@@ -0,0 +1,296 @@
+{
+    /// <summary>
+    /// An enum for all the supported diagnostics IPC command
+    /// </summary>
+    public enum DiagnosticsCommand
+    {
+        Dump,
+        EventPipe
+        Profiler
+    }
+}
+```
+
+### DiagnosticsIpcHelper
+This is a class that contains some utility methods for various purposes.
+```cs
+namespace Microsoft.Diagnostics.Client
+{
+    /// <summary>
+    /// A utility class that contain various helper methods to interact with the diagnostics IPC.
+    /// </summary>
+    public class DiagnosticsIpcHelper
+    {
+        /// <summary>
+        /// Get all the active processes that can be attached to.
+        /// </summary>
+        /// <returns>
+        /// IEnumerable of all the active process IDs.
+        /// </returns>
+        public static IEnumerable<int> GetActiveProcesses();
+
+        // The default path that the diagnostics client library looks for the IPC socket (default: /tmp)
+        public static string DefaultIpcSocketPath { get; set; } 
+
+        /// <summary>
+        /// Get all the supported commands by this version of the diagnostics IPC.
+        /// </summary>
+        /// <returns>
+        /// IEnumerable of all supported commands Command Code.
+        /// </returns>
+        public static IEnumerable<int> GetAvailableCommands();
+
+        /// <summary>
+        /// Convert a command name in the right format (CommandSet:Command) - i.e. EventPipe.CollectTracing - to corresponding command code.
+        /// </summary>
+        /// <returns>
+        /// Corresponding command code for the given command name. This may throw UnknownCommandException.
+        /// </returns>
+        public static int GetCommandCodeFromCommandName(string commandName);
+
+        /// <summary>
+        /// Convert a command code to its string representation
+        /// </summary>
+        /// <returns>
+        /// Corresponding command name for the given command code. This may throw UnknownCommandException
+        /// </returns>
+        public static string GetCommandNameFromCommandCode(int commandCode);
+
+    }
+}
+```
+
+### EventPipeCommandHandler
+This is a CommandHandler class for sending EventPipe commands across the diagnostics IPC channel.
+```cs
+namespace Microsoft.Diagnostics.Client
+{
+    /// <summary>
+    /// A class for handling EventPipe commands.
+    /// </summary>
+    public class EventPipeCommandHandler
+    {
+        public ulong SessionId { get; }  /// A ulong representing the current session ID. Defaults to ulong.Max
+        public uint CircularBufferSizeMB { get; set; }   /// The maximum buffer size in MB. Defaults to 
+        public EventPipeSerializationFormat SerializationFormat { get; set; }  /// Serialization format. Defaults to NetTrace
+        
+        public EventPipeCommandHandler(int processId)
+
+        /// <summary>
+        /// Add a provider to the EventPipe command
+        /// </summary>
+        public void AddProvider(
```

Comment:
I think this function would make more sense on a higher-level abstraction (see comment above about that).